pkgs/by-name/ly/lyra-cursors/package.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, stdenvNoCC
+, inkscape
+, xcursorgen
+, fetchFromGitHub
+, fetchurl
+, styles ? [ ]
+}:
+let
+  validStyles = [
+    "LyraB"
+    "LyraF"
+    "LyraG"
+    "LyraP"
+    "LyraQ"
+    "LyraR"
+    "LyraS"
+    "LyraX"
+    "LyraY"
+  ];
+
+  selectedStyles =
+    if (styles == [ ]) then
+      validStyles
+    else
+      let unknown = lib.subtractLists validStyles styles; in
+      if (unknown != [ ]) then
+        throw "Unknown style(s): ${lib.concatStringsSep " " unknown}"
+      else styles;
+
+  # This is a buildscript from a fork of the upstream repository which addresses several issues
+  # such as the fact that the style to build isn't hardcoded. We don't simply use this fork as
+  # source, as the upstream repository is what we want to track.
```

Comment:
Could we simply `substituteInPlace` the build script ourselves? It seems like the main change we need here is just to use `$1` for the theme name, and it would feel better to me than relying on a downstream that makes a bunch of changes to the build script. However, if those other changes are important then this may be okay. In that case, it may be better to `fetchpatch2` the commit URL (`https://github.com/KiranWells/Lyra-Cursors/commit/2735acb37a51792388497c666cc28370660217cb.patch?full_index=1`) and apply it directly to the source, rather than using a separate derivation like this.

Reply:
It would be a little much for a simple `substituteInPlace`, as that build script makes several other important changes, namely:
- Not dumping the build artifacts into a fixed output directory, but one that includes the theme name
- Adding a "Cursors" suffix to the theme name declared in `index.theme`, which appears to be convention with cursor themes to make them look distinct from icon themes
- Replacing the overcomplicated `find`-based solution to rendering the SVGs. This means we have a lot less Inkscape invocations and thus faster builds.

I see 3 options here:
- We include a custom patch that includes a boiled down version of these changes in the package
- We write a completely custom build script directly in the derivation, but this would get somewhat complicated.
- We fetch the patch from the fork as you suggested.

What would you suggest?